frontend/packages/console-shared/src/components/dynamic-form/templates.tsx

```diff
@@ -53,11 +54,15 @@ export const FieldTemplate: React.FC<FieldTemplateProps> = (props) => {
   const type = getSchemaType(schema);
   const [dependencyMet, setDependencyMet] = React.useState(true);
   React.useEffect(() => {
-    const { dependency } = getUiOptions(uiSchema ?? {}) as DependencyUIOption; // Type defs for this function are awful
+    const { dependency } = getUiOptions(uiSchema ?? {}) as UiSchemaOptionsWithDependency; // Type defs for this function are awful
     if (dependency) {
       setDependencyMet(
-        dependency.value ===
-          _.get(formContext.formData ?? {}, ['spec', ...(dependency.path ?? [])], '').toString(),
+        dependency?.controlFieldValue ===
+          _.get(
+            formContext.formData ?? {},
+            ['spec', ...(dependency?.controlFieldPath ?? [])],
+            '',
+          ).toString(),
```

Comment:
Updated the uiSchema dependency option to include the control field name (the end of the path) to make sorting easier, and updated property naming to be more descriptive while I was at it.